lib/client.js

```diff
@@ -1494,14 +1498,34 @@ function write (client, request) {
     method === 'PATCH'
   )
 
+  if (util.isFormDataLike(body)) {
+    if (!extractBody) {
+      extractBody = require('./fetch/body.js').extractBody
+    }
+
+    const [bodyStream, contentType] = extractBody(body)
+    if (request.contentType == null) {
```

Comment:
Not a critique of this PR but if the content-type isn't always set here the request will always be invalid. A server can't parse a multipart/form-data body without the boundary type, which is included in the header.

Reply:
I guess that's a user error then and it will fail?

Reply:
But the request will still work, the server won't be able to process it which can be hard to debug. I had someone ask me a few days ago why a request was failing server side and it ended up being because they were setting a `content-type: multipart/form-data` header manually.

Reply:
It seems like curl overwrites the header, which can make it even harder to debug as an equivalent request from undici will fail whereas curl won't. Probably better off for another issue, I was just reminded of it.